src/navigation/types.ts

```diff
@@ -0,0 +1,20 @@
+
+
+export interface IRouterConfig {
+    component: React.ComponentType<{
+        routes: IRouterConfig[]
+    }>,
+    path: string,
+    key?: string,
+    exact?: boolean,
+    routes?: IRouterConfig[],
+}
+
+export interface IRouterConfigWithoutRoutes {
```

Comment:
why do we need 2 router configs?
it should be possible to have just 1

Reply:
Have no idea how to omit it